assets/js/phoenix.js

```diff
@@ -473,62 +493,77 @@ export class Channel {
   isLeaving(){ return this.state === CHANNEL_STATES.leaving }
 }
 
-  /** Initializes the Socket
-   *
-   *
-   * For IE8 support use an ES5-shim (https://github.com/es-shims/es5-shim)
-   *
-   * @param {string} endPoint - The string WebSocket endpoint, ie, `"ws://example.com/socket"`,
-   *                                               `"wss://example.com"`
-   *                                               `"/socket"` (inherited host & protocol)
-   * @param {Object} opts - Optional configuration
-   * @param {string} opts.transport - The Websocket Transport, for example WebSocket or Phoenix.LongPoll.
-   *
-   * Defaults to WebSocket with automatic LongPoll fallback.
-   * @param {Function} opts.encode - The function to encode outgoing messages.
-   *
-   * Defaults to JSON:
-   *
-   * ```javascript
-   * (payload, callback) => callback(JSON.stringify(payload))
-   * ```
-   *
-   * @param {Function} opts.decode - The function to decode incoming messages.
-   *
-   * Defaults to JSON:
-   *
-   * ```javascript
-   * (payload, callback) => callback(JSON.parse(payload))
-   * ```
-   *
-   * @param {number} opts.timeout - The default timeout in milliseconds to trigger push timeouts.
-   *
-   * Defaults `DEFAULT_TIMEOUT`
-   * @param {number} opts.heartbeatIntervalMs - The millisec interval to send a heartbeat message
-   * @param {number} opts.reconnectAfterMs - The optional function that returns the millsec reconnect interval.
-   *
-   * Defaults to stepped backoff of:
-   *
-   * ```javascript
-   *  function(tries){
-   *    return [1000, 5000, 10000][tries - 1] || 10000
-   *  }
-   * ```
-   * @param {Function} opts.logger - The optional function for specialized logging, ie:
-   * ```javascript
-   * logger: (kind, msg, data) => { console.log(`${kind}: ${msg}`, data) }
-   * ```
-   *
-   * @param {number}  opts.longpollerTimeout - The maximum timeout of a long poll AJAX request.
-   *
-   * Defaults to 20s (double the server long poll timer).
-   *
-   * @param {Object}  opts.params - The optional params to pass when connecting
-   *
-   *
-  */
-export class Socket {
+let Encoder = {
```

Comment:
Since this has both `encode` and `decode` functions, should we call this `Serializer` or `Codec` instead?